### DISCUSSION TITLE
Get YAML key which contains dot.

### DISCUSSION THREAD

**Discussion by artemklevtsov** [redpanda-data/connect#2692] [Q&A] [answered]
Hi,

How can I get value from the field which name contains a dot?

to reproduce:

```yaml
# data.yaml

k1.k2: value
```

```blobl
# test.blobl

root.keys = file("./data.yaml").parse_yaml().keys()
root."v0" = file("./data.yaml").parse_yaml()."k1.k2" # works but I need dynamic variable
root."v1" = file("./data.yaml").parse_yaml().get("k1.k2")
root."v2" = file("./data.yaml").parse_yaml().get("k1").get("k2")
```

test:

```bash
❯ echo | benthos blobl --file test.blobl
{"keys":["k1.k2"],"v0":"value","v1":null,"v2":null}
```

**Comment by mihaitodor** [comment, marked as answer]
Hey @artemklevtsov, you can try `root.src = file("./data.yaml").parse_yaml().get("k1~1k2")`. The syntax is documented here: https://docs.redpanda.com/redpanda-connect/configuration/field_paths/

> The characters `~` (%x7E) and `.` (%x2E) have special meaning in Redpanda Connect paths. Therefore `~` needs to be encoded as `~0` and `.` needs to be encoded as `~1` when these characters appear within a key.

Alternatively, this should also work: `root.src = file("./data.yaml").parse_yaml()."k1.k2"`

**Reply by artemklevtsov** [reply]
thanks for the quick and clear answer.
